near/omni-prover/evm-prover-proxy/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{
+    env, ext_contract, near_bindgen, AccountId, Gas, PanicOnDefault, Promise, PromiseError,
+};
+use omni_types::evm_events::parse_evm_event;
+use omni_types::prover_args::EvmVerifyProofArgs;
+use omni_types::prover_result::{ProofKind, ProverResult};
+use omni_types::ChainKind;
+
+/// Gas to call verify_log_entry on prover.
+pub const VERIFY_LOG_ENTRY_GAS: Gas = Gas::from_tgas(50);
+
+#[ext_contract(ext_prover)]
+pub trait Prover {
+    #[result_serializer(borsh)]
+    fn verify_log_entry(
+        &self,
+        #[serializer(borsh)] log_index: u64,
+        #[serializer(borsh)] log_entry_data: Vec<u8>,
+        #[serializer(borsh)] receipt_index: u64,
+        #[serializer(borsh)] receipt_data: Vec<u8>,
+        #[serializer(borsh)] header_data: Vec<u8>,
+        #[serializer(borsh)] proof: Vec<Vec<u8>>,
+        #[serializer(borsh)] skip_bridge_call: bool,
+    ) -> bool;
+}
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct RainbowOmniProverProxy {
```

Comment:
Rainbow or EVM? It's better to make it consistent with the folder name. I think it's specific that this is RainbowProver specifically for Ethereum. For Aurora, for example, it would be different.

Reply:
7b69b4dc6d0ab5ca7caed6a68008b275689aa492